### PR TITLE
PP-61-add TeacherInfoPage

### PR DESCRIPTION
**Pull request by Tanyaa-a**
### Description
Implemented TeacherInfoPage
### Changes
added teacher's photo, info, video, teacher's available classes with scroll and portfolio images
### Screenshots
- large screen
Screen Shot 2024-09-06 at 5 54 56 PM

- medium Screen

Screen Shot 2024-09-07 at 2 39 37 PM

- Mobile
Screen Shot 2024-09-07 at 2 39 47 PM

### CODE REVIEW

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    <div className="">
+      <div className="flex flex-col md:flex-row justify-center md:justify-around md:bg-lightGreen">
+        <div className="flex flex-col bg-pureWhite p-4 lg:p-10 ml-4 md:ml-12 md:my-12 rounded md:w-1/2">
+          <div className="flex flex-row gap-4">
+            <div className="flex m-2">
+              {teacherInfo?.profileImageUrl && (
+                <img
+                  src={teacherInfo.profileImageUrl}
+                  alt={`${teacherInfo.firstName} ${teacherInfo.lastName}`}
+                  className="w-20 h-20 rounded-full"
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:67]: Please, fix the image. it displays bad

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                />
+              )}
+            </div>
+            <div className="flex justify-center items-center">
+              <h1 className="text-center font-roboto font-medium">
+                {teacherInfo?.firstName} {teacherInfo?.lastName}
+              </h1>
+            </div>
+          </div>
+
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:77]: Could you please add here information about teachers education, specialties and Experience

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            Teacher's classes
+          </p>
+
+          <div
+            className="flex flex-row gap-4 my-4 overflow-x-scroll"
+            style={{ scrollSnapType: 'x mandatory' }}
+          >
+            {classes.length > 0 ? (
+              classes.map((classItem) => (
+                <div
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:123]: Add link to this classInfo page

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+          <div className="flex mt-4 gap-4">
+            <button className="font-spartan bg-red hover:bg-pureWhite hover:text-red px-4 py-1 border-2 border-transparent hover:border-red text-white font-spartan font-semibold text-sm sm:text-lg rounded-md transition duration-300 ease-in">
+              Book lesson
+            </button>
+            <button className="font-spartan bg-pureWhite px-4 py-1 hover:bg-red hover:text-pureWhite hover-border-red text-red font-spartan font-semibold text-lg rounded-md border-2 border-red">
+              Send Message
+            </button>
+          </div>
+        </div>
+        <div className="flex flex-col justify-center mx-auto p-4 m-4 md:m-12 rounded w-10/12 md:w-1/2 bg-lightGreen">
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:87]: We need to show here Profile Video
      
      
        <div className="flex flex-col justify-center mx-auto p-4 m-4 md:m-12 rounded w-10/12 md:w-1/2 bg-lightGreen">
          {teacherInfo?.profileVideoUrl && (
              <div className="bg-lightGreen rounded ">
                <video
                  className=" w-full rounded "
                  controls
                  src={teacherInfo.profileVideoUrl}
                  type="video/mp4"
                >
                  Your browser does not support the video format.
                </video>
              </div>
            )}
        </div>

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            ) : (
+              <p className="text-center">
+                No classes available for this teacher.
+              </p>
+            )}
+          </div>
+        </div>
+      </div>
+
+      <div className="flex flex-col bg-pureWhite p-4 m-8 rounded">
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:174]: Add here portfolioVideos

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        {teacherInfo?.profilePortfolioImages &&
+        teacherInfo.profilePortfolioImages.length > 0 ? (
+          <div>
+            <p className="text-3xl font-spartan text-center font-medium mb-4">
+              Portfolio
+            </p>
+            <img
+              className="w-1/2 md:w-1/6 rounded-md"
+              src={teacherInfo.profilePortfolioImages[0].url}
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:183]: You display only one picture

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+              alt="Teacher Portfolio"
+            />
+          </div>
+        ) : (
```
Comment (mariyador) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:187]: You need to display portfolio videos

**Review by ofeterovskaya — COMMENTED** [review]
Tanya, could you please solve merge conflicts first?

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+  const { teacherId } = useParams();
+  const navigate = useNavigate();
+  const [teacherInfo, setTeacherInfo] = useState(null);
+  const [classes, setClasses] = useState([]);
+  const [classesError, setClassesError] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  useEffect(() => {
+    const fetchTeacherAndClasses = async () => {
```
Comment (JoHeyJo) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:19]: Get `getUserData` and `getClassesData` is lumped into one catch-all block, making it difficult to discern whether an error occurred while fetching the teacher data or the classes data. Ideally you want each API call to have separate error handling.

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 3 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx:170]: added

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 2 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx:103]: added

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 1 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx]: done

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 4 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx]: done

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 5 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx:220]: added

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx (hunk 6 above)
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx]: fixed

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+        const allClasses = response.classes;
+
+        const myClassIds = teacherData.data.myClasses.map((id) => id);
+        const filteredClasses = allClasses.filter((classItem) =>
+          myClassIds.includes(classItem._id)
+        );
+
+        if (filteredClasses.length === 0) {
+          setClassesError({
+            noClassesError: `This teacher hasn't added any classes yet.`,
```
[the remark this answers is not in the document]
Comment (Tanyaa-a) [reply src/components/pages/teacher-info-page/TeacherInformPage.jsx:34]: I got it

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 10]
```diff
@@ -21,26 +21,30 @@ const TeacherInfoPage = () => {
         const teacherData = await getUserData(teacherId, userData.token);
         setTeacherInfo(teacherData.data);
 
-        const response = await getClassesData();
-        const allClasses = response.classes;
+        try {
```
Comment (JoHeyJo) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:24]: This is a better approach. But, nesting makes this a little hard to read. Generally, you can have each action be their own function and then you can call it within the `useEffect`. That also decouples the user and class requests and allows them to work independent of each other.

File: src/components/pages/teacher-info-page/TeacherInformPage.jsx [hunk 11]
```diff
@@ -21,26 +21,30 @@ const TeacherInfoPage = () => {
...
-          myClassIds.includes(classItem._id)
-        );
+          const myClassIds = teacherData.data.myClasses.map((id) => id);
+          const filteredClasses = allClasses.filter((classItem) =>
+            myClassIds.includes(classItem._id)
+          );
 
-        if (filteredClasses.length === 0) {
+          if (filteredClasses.length === 0) {
+            return <p> This teacher hasn't added any classes yet</p>;
```
Comment (JoHeyJo) [inline src/components/pages/teacher-info-page/TeacherInformPage.jsx:34]: Clever approach. Much better.

**Review by JoHeyJo — APPROVED** [review]
Nice job cleaning things up!

**Review by ofeterovskaya — APPROVED** [review]
Perfect Job, Tanya!!! Thank you

**Pull request merged**